### ISSUE TITLE
`NoSuchElementException` while formatting a file

### ISSUE THREAD

**Issue by usommerl** [scalameta/sbt-scalafmt#287]
`scalafmt` throws a `NoSuchElementException` and is unable to format a certain file within my project. Unfortunately, I can't share the source code here. I'm not sure whether I can create a different file that triggers the same behaviour.

Used versions:
```
sbt          : v1.9.6
sbt-scalafmt : v2.5.2
scala        : v2.13.12
```

scalafmt.conf:
```
version = "3.0.0-RC3"

maxColumn = 140
align.preset = most
continuationIndent.defnSite = 2
rewrite.rules = [SortImports, AvoidInfix, PreferCurlyFors, RedundantBraces, RedundantParens]
runner.dialect = scala213source3
```

Here is the stack trace from the metals log file:
```
2023.10.25 10:32:47 ERROR scalafmt: /path/to/File.scala
java.util.NoSuchElementException: key not found: 1981584112128888851
        at scala.collection.immutable.BitmapIndexedMapNode.apply(HashMap.scala:674)
        at scala.collection.immutable.BitmapIndexedMapNode.apply(HashMap.scala:672)
        at scala.collection.immutable.HashMap.apply(HashMap.scala:132)
        at org.scalafmt.internal.FormatTokens.matching(FormatTokens.scala:72)
        at org.scalafmt.internal.FormatOps.orElse$1(FormatOps.scala:852)
        at org.scalafmt.internal.FormatOps.functionExpire(FormatOps.scala:867)
        at org.scalafmt.internal.Router.multilineSpaceSplit$1(Router.scala:654)
        at org.scalafmt.internal.Router.getSplitsImpl(Router.scala:685)
        at org.scalafmt.internal.Router.getSplits(Router.scala:2105)
        at org.scalafmt.internal.BestFirstSearch.$anonfun$routes$1(BestFirstSearch.scala:37)
        at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:576)
        at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:574)
        at org.scalafmt.internal.FormatTokens.foreach(FormatTokens.scala:15)
        at org.scalafmt.internal.BestFirstSearch.<init>(BestFirstSearch.scala:37)
        at org.scalafmt.internal.BestFirstSearch$.apply(BestFirstSearch.scala:334)
        at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$5(Scalafmt.scala:128)
        at scala.util.Try$.apply(Try.scala:210)
        at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$3(Scalafmt.scala:128)
        at scala.meta.parsers.Parsed.fold(Errors.scala:12)
        at scala.meta.parsers.Parsed.fold$(Errors.scala:11)
        at scala.meta.parsers.Parsed$Success.fold(Errors.scala:26)
        at org.scalafmt.Scalafmt$.doFormatOne(Scalafmt.scala:124)
        at org.scalafmt.Scalafmt$.doFormat(Scalafmt.scala:102)
        at org.scalafmt.Scalafmt$.formatCode(Scalafmt.scala:71)
        at org.scalafmt.Scalafmt$.format(Scalafmt.scala:52)
        at org.scalafmt.Scalafmt.format(Scalafmt.scala)
        at jdk.internal.reflect.GeneratedMethodAccessor92.invoke(null)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:568)
        at org.scalafmt.dynamic.ScalafmtReflect.$anonfun$tryFormat$1(ScalafmtReflect.scala:121)
        at scala.util.Try$.apply(Try.scala:210)
        at org.scalafmt.dynamic.ScalafmtReflect.tryFormat(ScalafmtReflect.scala:117)
        at org.scalafmt.dynamic.ScalafmtReflectConfig.tryFormat(ScalafmtReflectConfig.scala:95)
        at org.scalafmt.dynamic.ScalafmtDynamicSession.$anonfun$tryForceFormat$1(ScalafmtDynamicSession.scala:49)
        at scala.util.Success.flatMap(Try.scala:258)
        at org.scalafmt.dynamic.ScalafmtDynamicSession.tryForceFormat(ScalafmtDynamicSession.scala:49)
        at org.scalafmt.dynamic.ScalafmtDynamicSession.tryFormat(ScalafmtDynamicSession.scala:36)
        at org.scalafmt.dynamic.ScalafmtDynamicSession.format(ScalafmtDynamicSession.scala:18)
        at org.scalafmt.dynamic.ScalafmtDynamic.format(ScalafmtDynamic.scala:50)
        at scala.meta.internal.metals.FormattingProvider.runFormat(FormattingProvider.scala:147)
        at scala.meta.internal.metals.FormattingProvider.format(FormattingProvider.scala:118)
        at scala.meta.internal.metals.MetalsLspService.$anonfun$formatting$4(MetalsLspService.scala:1431)
        at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(Thread.java:833)
```

**Comment by usommerl** [comment]
I tried to format the same file with the standalone version of `scalafmt` and the same exception was thrown. Hence, this is not a bug within the sbt plugin. I will close the issue here and report it upstream.

**Comment by usommerl** [comment]
My bad! I was using a pretty outdated version of `scalafmt`. The problem is fixed in the latest release (`v3.7.5`).
